python/pyarrow/table.pxi

```diff
@@ -1437,6 +1451,21 @@ def chunked_array(arrays, type=None):
 
     if isinstance(arrays, Array):
         arrays = [arrays]
+    elif hasattr(arrays, "__arrow_c_stream__"):
+        if type is not None:
+            requested_type = type.__arrow_c_schema__()
+        else:
+            requested_type = None
+        capsule = arrays.__arrow_c_stream__(requested_type)
+        result = ChunkedArray._import_from_c_capsule(capsule)
+        if type is not None and result.type != type:
```

Comment:
If `type` is None then `requested_type` is None, so it shouldn't matter which of the two I am checking here. But since this if clause is to check for whether we still need to cast to `type` (and not `requested_type`, which is a capsule not a pyarrow data type), it seems more logical to check for `type is not None`.